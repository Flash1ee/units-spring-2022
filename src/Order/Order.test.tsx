jest.mock('../utils/getDate');
import React from 'react';
import { OrderComponent } from './Order';
import {getDate} from '../utils/getDate';
import {fakeOrders} from '../data/fakeOrders';
import {shallow, configure} from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Order.tsx', () => {
	describe('Order.tsx', () => {
		beforeEach(() => {
			(getDate as jest.Mock).mockReturnValue('Нажми на носик');
		});
	
		afterEach(() => {
			jest.clearAllMocks();
		});
	
		it('fakeOrders[0] - no items in the order - expected empty order', () => {
			const order = fakeOrders[0];
	
			const wrapper = shallow(<OrderComponent
				order={order}/>);
			expect(wrapper).toMatchSnapshot();
			expect(getDate).toBeCalled();
		});
		
		it('fakeOrders[1] - order with items - expect items on the page', () => {
			const order = fakeOrders[1];

			const wrapper = shallow(<OrderComponent order={order}/>);
			expect(wrapper).toMatchSnapshot();
			expect(getDate).toBeCalled();
		});
	
		
		test.each([
			{
				date: 5,
				shop: undefined, 
				items:[]
			},
			{
				date: undefined, 
				shop: 'amazon', 
				items:[1, 2]
			},
		])('OrderComponent({order: %s})', (order) => {
			const orderComponent = OrderComponent({order: order}); 
			expect(orderComponent).toBeNull();
		});
	
		
	});
});
