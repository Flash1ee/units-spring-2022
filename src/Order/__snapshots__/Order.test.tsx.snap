// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.tsx Order.tsx fakeOrders[0] - no items in the order - expected empty order 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Сбереги Мега Маркер
    </span>
    <span
      id="date"
    >
      Нажми на носик
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;

exports[`Order.tsx Order.tsx fakeOrders[1] - order with items - expect items on the page 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      Нажми на носик
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;
